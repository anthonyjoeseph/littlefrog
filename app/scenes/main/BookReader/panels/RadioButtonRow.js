// app/BookReader/panels/RadioButtonRow.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Animated,
  TouchableWithoutFeedback
} from 'react-native';

import SubPanel from './SubPanel';
import { RadioButtons } from 'react-native-radio-buttons';

class RadioButtonRow extends Component{
  constructor(props){
    super(props);

    this.state = {
      selectedOption: props.options[0]
    };

    this.setSelectedOption = this.setSelectedOption.bind(this);
    this.renderOption = this.renderOption.bind(this);
    this.renderContainer = this.renderContainer.bind(this);
  }

  componentWillMount(){
    this.props.onSelection(this.state.selectedOption);
  }

  setSelectedOption(selectedOption){
    this.setState({
      selectedOption
    });
    this.props.onSelection(selectedOption);
  }

  renderOption(option, selected, onSelect, index){
    const containerStyle = selected ? { backgroundColor: 'yellow' } : { backgroundColor: 'white' };
    const style = selected ? { fontWeight: 'bold' } : {};

    return (
      <TouchableWithoutFeedback
        onPress={onSelect}
        key={index}>
        <View style={[containerStyle,
          {
           flex:1,
           flexDirection: 'row',
           justifyContent: 'center',
           alignItems: 'center'
          }]}>
          <Text style={style}>{option}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }

  renderContainer(optionNodes){
    return <View style={{
      flex:1,
      flexDirection: 'row',
      justifyContent: 'space-around',
      alignItems: 'stretch'
    }}>{optionNodes}</View>;
  }

  render(){
      return (
        <RadioButtons
          options={ this.props.options }
          onSelection={ this.setSelectedOption }
          selectedOption={ this.state.selectedOption }
          renderOption={ this.renderOption }
          renderContainer={ this.renderContainer }
        />
      );
  }
}

export default RadioButtonRow;
