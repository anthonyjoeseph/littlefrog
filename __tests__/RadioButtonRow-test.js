// __tests__/DismissableFullscreenPopup-test.js
import 'react-native';
import React from 'react';
import RadioButtonRow from '../app/scenes/main/BookReader/panels/RadioButtonRow';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <RadioButtonRow
      options={["Testing1", "Testing2", "Testing3"]}
      onSelection={ (selection) => { } }
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
