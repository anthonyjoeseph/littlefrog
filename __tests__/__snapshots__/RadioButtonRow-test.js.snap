exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "yellow",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
        }
      }>
      Testing1
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}>
      Testing2
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}>
      Testing3
    </Text>
  </View>
</View>
`;
